llvm/lib/Target/AIE/AIEMachineScheduler.cpp

```diff
@@ -149,6 +152,14 @@ std::vector<AIE::MachineBundle> computeAndFinalizeBundles(SchedBoundary &Zone) {
       if (!SU)
         continue;
       unsigned EmitCycle = Zone.isTop() ? SU->TopReadyCycle : SU->BotReadyCycle;
+
+      if (!ComputeSlots && EmitCycle < Bundles.size()) {
+        // The pre-RA scheduler can actually re-order copies and immediate
+        // moves, disregarding the emission cycle.
+        // See GenericScheduler::reschedulePhysReg().
```

Comment:
This leaves the possibility that EmitCycle > Bundles.size() is incorrect? Perhaps detail the comment that this is not a problem in preRA?
Also, perhaps assert that we are preRA if it occurs?